engine/src/main/java/org/terasology/entitySystem/entity/EntityRef.java

```diff
@@ -73,11 +74,17 @@
     public abstract boolean isPersistent();
 
     /**
-     * Sets whether this entity should be saved
+     * To simplify things for the {@link StorageManager} the persistent property of entities can't be changed after
+     * creation. The {@link EntityManager#newBuilder()}} methods can be used to construct non persistent entities.
      *
-     * @param persistent
+     * The existence of this private method will ensure that someone who wants to add this setter again will
+     * read the above comment.
      */
-    public abstract void setPersistent(boolean persistent);
+    @SuppressWarnings("unused")
+    @Deprecated
```

Comment:
What does EntityBuilder have to do with anything? It doesn't inherit EntityRef. It is a builder, it is expected it can set values that may be immutable later.

EntityRef defines the API for EntityRefs. If you remove the setPersistent method, it is gone. No-one can call it. You can put the comment on it being immutable in the class javadoc or the javadoc for isPersistent().

Overall this method doesn't achieve anything over a bit of javadoc (and being obnoxious IMO, since it will trigger FindBugs and CheckStyle warnings). A child class can still add the method, without any problem for instance. One could add a method with a slightly different name without issue.

And yes, I agree that good code is hard to break, but the method doesn't even qualify as code, and offers no protection since there are other ways to modify the persistence flag. Better would be putting protection in the EntityMetadata component to prevent the persistence field being modified - make it protected or private, and set during construction only.